frontend/src/features/staffSlice.ts

```diff
@@ -28,22 +28,20 @@ export const staffSlice = createSlice({
 
 export const { set } = staffSlice.actions;
 
-const fetchWithAuth = (
-  url: string,
-  method: string,
-  getToken: () => string
-): AppThunk => (dispatch) =>
-  fetch(url, { method, headers: { Authorization: getToken() } })
+export const fetchStaff = (token: String | undefined): AppThunk => (dispatch) =>
+  get(STAFF_FETCH_URL, token)
     .then((response) => response.json())
-    .then((response) => dispatch(set(response)));
+    .then((response) => dispatch(set(response)))
+    .catch((err) => console.error(err));
 
-export const fetchStaff = fetchWithAuth(
-  STAFF_FETCH_URL,
-  'GET',
-  getAuthenticationToken
-);
-export const deleteStaff = (id: string) =>
-  fetchWithAuth(STAFF_DELETE_URL + id, 'PUT', getAuthenticationToken);
+export const deleteStaff = (
+  id: string,
+  token: String | undefined
+): AppThunk => (dispatch) =>
+  put(STAFF_DELETE_URL + id, token)
+    .then((response) => response.json())
+    .then((response) => dispatch(set(response)))
+    .catch((err) => console.error(err));
```

Comment:
I'm OK with `console.error`, but we can squelch the warning here by using `toastr.error` instead. The behaviour will be different and the user will get a much more noticeable feedback, so not sure if that's what we want.

Reply:
Yeah okay this is a better option I think. I'll use this for now which will at least avoid any warnings, we will just need to make sure that the error that is shown makes sense for the situation.